oauth2/handler.go

```diff
@@ -1154,3 +1207,175 @@ func (h *Handler) logOrAudit(err error, r *http.Request) {
 		x.LogAudit(r, err, h.r.Logger())
 	}
 }
+
+// swagger:route POST /credentials oAuth2 createVerifiableCredential
+//
+// # Issues a Verifiable Credential
+//
+// This endpoint creates a verifiable credential that attests that the user
+// authenticated with the provided access token owns a certain public/private key
+// pair.
+//
+// More information can be found at
+// https://openid.net/specs/openid-connect-userinfo-vc-1_0.html.
+//
+//	Consumes:
+//	- application/json
+//
+//	Schemes: http, https
+//
+//	Responses:
+//	  200: verifiableCredentialResponse
+//	  400: verifiableCredentialPrimingResponse
+//	  default: errorOAuth2
+func (h *Handler) createVerifiableCredential(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	session := NewSessionWithCustomClaims("", h.c.AllowedTopLevelClaims(ctx))
+	accessToken := fosite.AccessTokenFromRequest(r)
+	tokenType, _, err := h.r.OAuth2Provider().IntrospectToken(ctx, accessToken, fosite.AccessToken, session)
+
+	if err != nil {
+		h.r.Writer().WriteError(w, r, errorsx.WithStack(err))
+		return
+	}
+	if tokenType != fosite.AccessToken {
+		h.r.Writer().WriteError(w, r, errorsx.WithStack(fosite.ErrInvalidRequest.WithHintf("the token is not an access token")))
+		return
+	}
+
+	var request CreateVerifiableCredentialRequestBody
+	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
+		h.r.Writer().WriteError(w, r, errorsx.WithStack(err))
+		return
+	}
+
+	if request.Format != "jwt_vc_json" {
+		h.r.Writer().WriteError(w, r, errorsx.WithStack(fosite.ErrInvalidRequest.WithHintf("the format is not supported")))
+		return
+	}
+	if request.Proof == nil {
+		// Handle priming request
+		nonceLifespan := h.r.Config().GetVerifiableCredentialsNonceLifespan(ctx)
+		nonceExpiresIn := time.Now().Add(nonceLifespan).UTC()
+		nonce, err := h.r.OAuth2Storage().NewNonce(ctx, accessToken, nonceExpiresIn)
+		if err != nil {
+			h.r.Writer().WriteError(w, r, errorsx.WithStack(err))
+			return
+		}
+		w.WriteHeader(http.StatusBadRequest)
+		h.r.Writer().Write(w, r, &VerifiableCredentialPrimingResponse{
```

Comment:
I think that `Writes` sets the status code. I believe that herodot has a method to override the status code - use that instead